mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -480,8 +480,8 @@ class RegionBuilderHelper {
     llvm_unreachable("unsupported type conversion function");
   }
 
-  void yieldOutputs(ValueRange values) {
-    OpBuilder builder = getBuilder();
+  void yieldOutputs(OpBuilder builder, ValueRange values) {
+    builder.setInsertionPointToEnd(&block);
```

Comment:
What's the difference here exactly with before? Looking at the implementation of `getBuilder()` I'm not spotting it immediately.

Reply:
In the previous one the builder is being constructed from the block's argument context  `block.getArgument(0).getContext()` , here is not the case It is being passed as is.
I really wondered why should this result a memory leak. Therefore, would be glad for your review/advice :) .